app/models/doctor.rb

```diff
@@ -1,3 +1,9 @@
 class Doctor < ApplicationRecord
   belongs_to :hospital
+  has_many :doctor_patients
+  has_many :patients, through: :doctor_patients
+
+  def self.patient_count(hospital_id)
+    Doctor.joins(:patients, :hospital).where(hospital: {id: "#{hospital_id}"}).select("doctors.name, doctors.id, COUNT(patients.id) AS patient_count").group("doctors.id").order("patient_count DESC")
+  end
```

Comment:
We're not really leaving PR feedback but I just wanted to point out that this is way too much work. This is equivalent to this:

```ruby
def patient_count
  self.patients.distinct.count
end
```